### PR TITLE
feat: add trace export sample via http/protobuf

### PR DESCRIPTION
**Pull request by psx95**
Very similar to existing samples, except that the OTLP exporter used is the one for http with protobuf.

### Testing
 - `npm run lint`
 - `npm run start-http-proto`

The existing client can be re-used for indefinite export.

### CODE REVIEW

**Review by aabmass — COMMENTED** [review]
Are we going to fold the https://github.com/GoogleCloudPlatform/opentelemetry-operations-js/tree/main/samples/otlptracehttp sample in here as well or did you want to do a follow up PR?

File: samples/otlptraceexport/src/app-http-proto-export.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+import {NodeSDK} from '@opentelemetry/sdk-node';
+import {OTLPTraceExporter} from '@opentelemetry/exporter-trace-otlp-proto';
+import {AuthClient, GoogleAuth} from 'google-auth-library';
+
+const PORT = parseInt(process.env.PORT || '8080');
+const app = express();
+
+async function getAuthenticatedClient(): Promise<AuthClient> {
+  const auth: GoogleAuth = new GoogleAuth({
```
Comment (aabmass) [inline samples/otlptraceexport/src/app-http-proto-export.ts:26]: Seems to have been last PR
```suggestion
  const auth = new GoogleAuth({
```

File: samples/otlptraceexport/src/app-http-proto-export.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    scopes: 'https://www.googleapis.com/auth/cloud-platform',
+  });
+  return await auth.getClient();
+}
+
+// Express App that exports traces over HTTP with protobuf
+async function main() {
+  const authenticatedClient: AuthClient = await getAuthenticatedClient();
```
Comment (aabmass) [inline samples/otlptraceexport/src/app-http-proto-export.ts:34]: Any reason to explicitly annotate here?

**Review by aabmass — APPROVED** [review]

**Review by damemi — APPROVED** [review]

**Comment by psx95** [comment]
> Are we going to fold the https://github.com/GoogleCloudPlatform/opentelemetry-operations-js/tree/main/samples/otlptracehttp sample in here as well or did you want to do a follow up PR?

Was planning to do that in a follow-up PR

File: samples/otlptraceexport/src/app-http-proto-export.ts (hunk 2 above)
Comment (psx95) [reply samples/otlptraceexport/src/app-http-proto-export.ts:34]: Nothing other than making it explicit for the reader. I can remove it to adhere to the Typescript style guide.

**Pull request merged**